src/unity/python/turicreate/visualization/plot.py

```diff
@@ -139,21 +150,35 @@ def save(self, filepath, include_data=True):
         elif filepath.endswith(".png") or filepath.endswith(".svg"):
             # save as png/svg, but json first
             spec = self._get_vega(include_data = True)
-            extension = filepath[-3:]
+            EXTENSION_START_INDEX = -3
+            extension = filepath[EXTENSION_START_INDEX:]
             temp_file_tuple = _mkstemp()
             temp_file_path = temp_file_tuple[1]
             with open(temp_file_path, 'w') as fp:
                 _json.dump(spec, fp)
             dirname = _os.path.dirname(__file__)
             relative_path_to_vg2png_vg2svg = "../vg2" + extension
-            absolute_path_to_vg2png_vg2svg = _os.path.join(
-                dirname, relative_path_to_vg2png_vg2svg)
-            _os.system("node " + absolute_path_to_vg2png_vg2svg + " " + temp_file_path + " " + filepath)
+            absolute_path_to_vg2png_vg2svg = _os.path.join(dirname, 
+                relative_path_to_vg2png_vg2svg)
+            (exitcode, stdout, stderr) = _run_cmdline("node " + 
+                absolute_path_to_vg2png_vg2svg + " " 
+                + temp_file_path + " " + filepath)
+            if exitcode == _NODE_NOT_FOUND_ERROR_CODE:
+                # user doesn't have node installed
+                raise NameError("Node.js not found. Saving as PNG and SVG" +
```

Comment:
Looking again at this, I don't think `NameError` is appropriate for this use -- it's supposed to mean an undeclared identifier (local or global variable) is being referenced. Maybe this should just be `RuntimeError`.